src/esp/bindings_js/modules/replay_viewer.js

```diff
@@ -0,0 +1,373 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+/*global Module */
+import SimEnv from "./simenv_embind";
+
+/**
+ * ReplayViewerTask class
+ */
+class ReplayViewerTask {
+  // PUBLIC methods.
+
+  /**
+   * Create ReplayViewer task.
+   * @param {SimEnv} sim - simulator
+   * @param {Object} components - dictionary with status and canvas elements
+   */
+  constructor(sim, components) {
+    this.sim = sim;
+    this.components = components;
+
+    this.actions = [
+      { name: "moveForward", key: "w", keyCode: 87 },
+      { name: "moveBackward", key: "s", keyCode: 83 },
+      { name: "strafeLeft", key: "a", keyCode: 65 },
+      { name: "strafeRight", key: "d", keyCode: 68 },
+      { name: "turnLeft", key: "ArrowLeft", keyCode: 37 },
+      { name: "turnRight", key: "ArrowRight", keyCode: 39 },
+      { name: "lookUp", key: "ArrowUp", keyCode: 38 },
+      { name: "lookDown", key: "ArrowDown", keyCode: 40 }
+    ];
+  }
+
+  /**
+   * Initialize the task. Should be called once.
+   */
+  init() {
+    this.bindKeys();
+
+    this.activeActions = [];
+    this.isPaused = false;
+
+    this.initAgentState();
+
+    this.frameCounter = 0;
+    this.replayStepFunction = setInterval(() => {
+      if (this.player) {
+        if (!this.isPaused) {
+          var currIndex = this.player.getKeyframeIndex();
+          var newIndex =
+            currIndex == this.player.numKeyframes - 1 ? 0 : currIndex + 1;
+          this.player.setKeyframeIndex(newIndex);
+        }
+      }
+
+      for (let action of this.activeActions) {
+        this.handleAction(action);
+      }
+
+      this.render();
```

Comment:
Notice the Replay Viewer doesn't call sim.step. It doesn't use the agent action API.